File: Top 50 Coding Interview Questions/26. Write a program to check whether a character is a vowel or consonant/CharecterVowelOrConsent.cpp
#include <iostream>
#include <cctype> // For tolower function
using namespace std;

int main() {
    char ch;
    cout << "Enter a character: ";
    cin >> ch;

    // Convert the character to lowercase for simplicity
    ch = tolower(ch);

    // Check if the character is an alphabet
    if (isalpha(ch)) {
        // Check if the character is a vowel
        if (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u') {
            cout << ch << " is a vowel." << endl;
        } else {
            cout << ch << " is a consonant." << endl;
        }
    } else {
        cout << ch << " is not an alphabet." << endl;
    }

    return 0;
}

File: Top 50 Coding Interview Questions/26. Write a program to check whether a character is a vowel or consonant/CharecterVowelOrConsent.py
# Function to check whether a character is a vowel or consonant
def check_character(ch):
    # Convert the character to lowercase for simplicity
    ch = ch.lower()

    # Check if the character is an alphabet
    if ch.isalpha():
        # Check if the character is a vowel
        if ch in 'aeiou':
            print(f"{ch} is a vowel.")
        else:
            print(f"{ch} is a consonant.")
    else:
        print(f"{ch} is not an alphabet.")

# Input from the user
char = input("Enter a character: ")

# Validate input length
if len(char) == 1:
    check_character(char)
else:
    print("Please enter only one character.")
